pkg/cvo/sync_worker.go

```diff
@@ -58,6 +58,10 @@ type PayloadRetriever interface {
 	RetrievePayload(ctx context.Context, desired configv1.Update) (PayloadInfo, error)
 }
 
+type ClusterProfileRetriever interface {
+	RetrieveClusterProfile(ctx context.Context) string
```

Comment:
You should return an error in this function. It's probably safer to fail in the case you can't read the profile than to assume "default"

Reply:
Just to clarify, you can check the error for IsNotFound and return default... however other errors probably indicate an api server error and you should probably report.

Reply:
yes you're right, it's better now.